index.bs

```diff
@@ -292,6 +295,43 @@ partial interface NeuralNetworkContext {
     Computes a 2-D convolution given input and filter 4-D tensors.
 </div>
 
+### gemm ### {#api-neuralnetworkcontext-gemm}
+<script type=idl>
+partial interface NeuralNetworkContext {
+  static Operand gemm(NeuralNetworkContext nn, Operand a, Operand b, optional Operand c = null, 
```

Comment:
I don't know IDL conventions enough, but is it possible to make this a non-static method? It seems to me that has the advantage of preserving the style in which operations are specified (that is, no difference between how you invoke "gemm" and "matmul" from user's perspective.)

Reply:
Note that we can always have a separate static-method, if needed, to define the expansion.

Reply:
It's always possible to make this a regular method instead of a static one. The WebIDL convention allows both. At the conceptual level though, a static method is considered part of the interface while a regular one is part of an instance of the interface (the object). It implies that a static method is in fact a helper function that requires no state within the interface instance to function properly.

From the caller's point of view, calling into a static method is like calling into a function, and not to an interface object's method. So the semantic is a bit different. 

I propose this distinction in the API to highlight the fact that there are regular operations of the interface (small ops) vs. helper functions to construct subgraphs made of regular operations (big ops).

A nice twist here though is that WebIDL allows static methods to be implementable by the interface implementer. This effectively means the browsers can now override the implementations of these helper functions to their own choosing. For instance, they can implement gemm as a fused operation at the OS layer, etc.

Reply:
I dropped the static method for the gemm operation and instead creating a non-normative note section in the operation's description. Thanks @rafaelcintron for the WebGL sample on non-normative section of a standard spec. @gramalingam please would you have a look at my new commit and let me know if it addresses your concern.